src/components/PopupWithForm.js

```diff
@@ -20,8 +21,14 @@ export default class PopupWithForm extends Popup {
 
   reset() {
     this._form.reset();
-    // the review comment that was sent got lost and would not popup for this section so i couldnt see what needed to be fixed
-    //i will remove these comments when it gets checked again i had to try to remember what was said by memory
+  }
+
```

Comment:
Needs correction (line 39): please don't reset the form right after calling the submit handler, you should do that in index.js after the successful api response in `then` block, so the form won't be cleared if an error occurred